src/list/list.test.tsx

```diff
@@ -0,0 +1,114 @@
+import React from "react";
+import {render, screen} from "@testing-library/react";
+import "@testing-library/jest-dom";
+import {create} from "react-test-renderer";
+
+import {testA11y} from "../core/utils/test/testUtils";
+import List, {ListProps} from "./List";
+import ListItem from "./item/ListItem";
+
+describe("<List />", () => {
+  const listItems = ["Hipo", "Labs"];
+  const defaultListProps: ListProps = {
+    testid: "list",
+    items: listItems,
+    children: (item, testId, index) => (
+      <ListItem key={`list-item-${index}`} testid={testId}>
+        <span>{item}</span>
+      </ListItem>
+    )
+  };
+
+  it("should render correctly", () => {
+    render(<List {...defaultListProps} />);
+  });
+
+  it("should match snapshot", () => {
+    const tree = create(<List {...defaultListProps} />).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("should pass a11y test", async () => {
+    const {container} = render(<List {...defaultListProps} />);
+
+    await testA11y(container);
+  });
+
+  it("should render children correctly", () => {
+    const {getByTestId} = render(<List {...defaultListProps} />);
+
+    expect(getByTestId(defaultListProps.testid!).children).toHaveLength(listItems.length);
+
+    for (let index = 0; index < listItems.length; index++) {
+      const listItemContent = getByTestId(`list.item-${index}`);
+
+      expect(getByTestId(defaultListProps.testid!).children[index]).toEqual(
+        listItemContent
+      );
+    }

```

Comment:
This is heavily dependant on the implementation details. Please try to improve and keep it in line with the [guidelines](https://testing-library.com/docs/guiding-principles/) of testing-library

Reply:
To avoid being dependent to implementation details, i added `listitem` role to list items so we can get them from their roles. We dont need testid's anymore. 
https://github.com/Hipo/react-ui-toolkit/pull/103/commits/b982cbe8868a164995e2a82138741d6a26052e2a